Fix Employer Branch ID Issue and Implement Employer Bank Account Update

PramithaMJ opened the pull request:
# Pull Request: Fix Employer Branch ID Issue and Implement Employer Bank Account Update

## Title
Fix Employer Branch ID Issue and Implement Employer Bank Account Update Endpoint

## Description

This pull request addresses the issue where the Employer Branch ID incorrectly shows as 0 when retrieved from the database. Additionally, it implements a new endpoint for updating employer bank account details with a specified employer ID.

### Changes Made

#### Employer Branch ID Issue Fix:
- Updated the logic to correctly retrieve the Employer Branch ID from the database.
- Fixed the issue where the Employer Branch ID was incorrectly showing as 0 by ensuring the correct database retrieval and mapping.

#### New Endpoint Implementation:
- Added a new PUT endpoint `/updateEmployerBankAccountDetailsWithId/{employerId}` to update employer bank account details with a specified employer ID.
- Implemented the endpoint logic to update employer bank account details based on the provided employer ID and request body data.
- Utilized transaction management to ensure data consistency and integrity during the update operation.

#### API Test and Screenshots:
- Conducted API testing to validate the functionality of the new endpoint.
- Added screenshots of the API tests to the documentation for reference and verification.

### Impact Analysis:
- These changes may affect the behavior of the Employer service and related API endpoints.
- The issue fix ensures the correct retrieval and representation of Employer Branch ID, improving data accuracy.
- The new endpoint provides functionality to update employer bank account details, enhancing the application's features and usability.

## Affected Files

- EmployerController.java
- EmployerService.java
- EmployerMapper.java
- EmployerUpdateBankAccountDTO.java
- API test screenshots

## Test Plan

- Verify the fix for the Employer Branch ID issue by retrieving employer data from the database.
- Test the new `/updateEmployerBankAccountDetailsWithId/{employerId}` endpoint with various scenarios, including valid and invalid input data.
- Validate the functionality and response of the updated API endpoints through thorough testing.
- Review API test screenshots to ensure accurate testing coverage and results.

## Related Issues

- Fixes #86 : Addresses the Employer Branch ID issue and adds the new employer bank account update endpoint.

## ScreenShots
<img width="1329" alt="Screenshot 2024-05-09 at 1 53 04 PM" src="https://github.com/Life-Pill/pharmacy-pos-backend/assets/123730262/54c0ef8f-45d8-4bf6-b1dd-16f9aa5315fe">
<img width="1310" alt="Screenshot 2024-05-09 at 4 07 23 PM" src="https://github.com/Life-Pill/pharmacy-pos-backend/assets/123730262/6c50814f-8ca3-4ef6-88cd-97d148c7a3cf">


## Checklist

- [x] Fixed the Employer Branch ID issue by updating the retrieval and mapping logic.
- [x] Implemented the `/updateEmployerBankAccountDetailsWithId/{employerId}` endpoint with the required functionality.
- [x] Conducted API testing to validate the correctness and effectiveness of the updated endpoints.
- [x] Added screenshots of API tests to the documentation for reference and verification.

## Conclusion

Thank you for reviewing this pull request. By addressing the Employer Branch ID issue and adding the new employer bank account update endpoint, we aim to improve the accuracy and functionality of the application. Please review the changes and provide any feedback or suggestions for further improvement.

Best regards,
Pramitha Jayasooriya
Backend Developer at LifePill
https://Pramithamj.me


## Donation

***If you like what I do, maybe consider buying me a coffee***

<a href="https://buymeacoffee.com/lpramithamm"><img src="https://cdn.buymeacoffee.com/buttons/v2/default-red.png" alt="Buy Me A Coffee" style="height: 35px !important; width: 120px !important;"></a>

***

TheTharz reviewed (changes requested):
# Request Changes - Fix Employer Branch ID Issue and Implement Employer Bank Account Update

**Pull Request:** Fix Employer Branch ID Issue and Implement Employer Bank Account Update Endpoint

## Reviewer Information
- **Name:** Tharindu Jayawadhana
- **Position:** Frontend Developer at LifePill

---

## Summary
This pull request addresses two main changes: fixing the Employer Branch ID issue and implementing a new endpoint for updating employer bank account details. The changes include updating logic to retrieve the Employer Branch ID correctly and adding a new PUT endpoint to update employer bank account details.

---

## Review Feedback
- **Employer Branch ID Issue Fix:** The changes to fix the Employer Branch ID issue seem appropriate and necessary.
- **New Endpoint Implementation:** However, the addition of a new endpoint to update employer bank account details does not align with the required functionality. Instead, the goal is to update the cashier creation endpoint to return the ID of the created cashier.

---

## Proposed Changes
1. **Update Cashier Creation Endpoint:**
   - Modify the focus of this pull request to update the cashier creation endpoint instead of the employer bank account update endpoint.
   - Ensure that when a cashier is created, the endpoint returns the ID of the created cashier.

---

## Test Plan
- Verify the updated cashier creation endpoint to ensure it returns the ID of the created cashier.
- Conduct thorough testing of the modified endpoint with various scenarios to validate functionality.
- Update API test screenshots in the documentation to reflect the changes made to the cashier creation endpoint.

---

## Conclusion
Thank you for the detailed pull request and explanations provided. However, I request changes to update the cashier creation endpoint as described above. This adjustment will better align the changes with the project requirements.

Best regards,
Tharindu Jayawadhana
Frontend Developer at LifePill

PramithaMJ commented:
# Pull Request: Update Cashier Creation Endpoint to Return Created Cashier ID

## Title
Update Cashier Creation Endpoint to Return Created Cashier ID

## Description

This pull request addresses the feedback provided by the reviewer regarding the previous changes. Instead of implementing a new endpoint for updating employer bank account details, the focus has been shifted to updating the cashier creation endpoint to return the ID of the created cashier. 

### Changes Made

#### Cashier Creation Endpoint Update:
- Modified the existing cashier creation endpoint logic to return the ID of the created cashier upon successful creation.
- Ensured that the endpoint now correctly returns the ID of the created cashier as required by the project specifications.

#### API Test Updates:
- Conducted thorough testing of the modified cashier creation endpoint with various scenarios to validate its functionality.
- Updated the API test screenshots in the documentation to reflect the changes made to the cashier creation endpoint.

### Impact Analysis:
- These changes have a direct impact on the functionality of the cashier creation endpoint, ensuring it aligns with the project requirements.
- By returning the ID of the created cashier, the endpoint provides valuable information for further operations and data management.

## Affected Files

- EmployerController.java
- API test screenshots

## ScreenShot

<img width="1247" alt="Screenshot 2024-05-09 at 4 45 19 PM" src="https://github.com/Life-Pill/pharmacy-pos-backend/assets/123730262/2c8f615c-1d2f-44a1-b99d-7c3bd8d49b02">

## Test Plan

- Verify the updated cashier creation endpoint functionality by creating new cashiers and checking for the returned ID.
- Conduct thorough testing of the modified endpoint with various scenarios, including valid and invalid input data.
- Review API test screenshots to ensure accurate testing coverage and results.

## Conclusion

Thank you for providing detailed feedback on the previous changes. This pull request updates the cashier creation endpoint as requested, ensuring it returns the ID of the created cashier. Please review the changes and confirm if they align with the project requirements.

Best regards,
Pramitha Jayasooriya
Backend Developer at LifePill
https://pramithamj.me


## Donation

***If you like what I do, maybe consider buying me a coffee***

<a href="https://buymeacoffee.com/lpramithamm"><img src="https://cdn.buymeacoffee.com/buttons/v2/default-red.png" alt="Buy Me A Coffee" style="height: 35px !important; width: 120px !important;"></a>

***

TheTharz reviewed (approved):
# Pull Request Review - Update Cashier Creation Endpoint to Return Created Cashier ID

**Pull Request:** Update Cashier Creation Endpoint to Return Created Cashier ID

## Reviewer Information
- **Name:** Tharindu Jayawadhana
- **Position:** Frontend Developer at LifePill

---

## Summary
This pull request focuses on updating the cashier creation endpoint to return the ID of the created cashier, addressing feedback from the previous review.

---

## Review Feedback
The changes made in this pull request align well with the required functionality. Shifting the focus to update the cashier creation endpoint to return the ID of the created cashier is in line with project specifications.

---

## Proposed Changes
No further changes are required as the pull request effectively updates the cashier creation endpoint to meet project requirements.

---

## Test Plan
- Verify the updated cashier creation endpoint to ensure it returns the ID of the created cashier.
- Conduct thorough testing of the modified endpoint with various scenarios to validate functionality.
- Update API test screenshots in the documentation to reflect the changes made to the cashier creation endpoint.

---

## Conclusion
Thank you for addressing the feedback and updating the cashier creation endpoint as required. The changes made in this pull request align with project specifications and improve the functionality of the endpoint. 

Best regards,
Tharindu Jayawadhana
Frontend Developer at LifePill
